database/repository/exchange/exchange.go

```diff
@@ -0,0 +1,204 @@
+	exchangeCache = cache.New(10)
+)
+
+type Details struct {
+	UUID uuid.UUID
+	Name string
+}
+
+// One returns one exchange by Name
+func One(in string) (out *Details, err error) {
+	if database.DB.SQL == nil {
+		return nil, database.ErrDatabaseSupportDisabled
+	}
+
+	out = new(Details)
+	whereQM := qm.Where("name = ?", in)
+	if repository.GetSQLDialect() == database.DBSQLite3 {
+		ret, err := modelSQLite.Exchanges(whereQM).One(context.Background(), database.DB.SQL)
+		if err != nil {
+			return nil, err
+		}
+		out.Name = ret.Name
+		out.UUID, err = uuid.FromString(ret.ID)
+		if err != nil {
+			return out, err
+		}
+	} else {
+		ret, err := modelPSQL.Exchanges(whereQM).One(context.Background(), database.DB.SQL)
+		if err != nil {
+			return nil, err
+		}
+		out.Name = ret.Name
+		out.UUID, _ = uuid.FromString(ret.ID)
+	}
+
+	return
+}
+
+// OneByUUID returns one exchange by UUID
+func OneByUUID(in uuid.UUID) (*modelPSQL.Exchange, error) {
+	if database.DB.SQL == nil {
+		return nil, database.ErrDatabaseSupportDisabled
+	}
+
+	return modelPSQL.FindExchange(context.Background(), database.DB.SQL,
+		in.String())
+}
+
+// Insert writes a single entry into database
+func Insert(in Details) {
+	if database.DB.SQL == nil {
+		return
+	}
+
+	ctx := boil.SkipTimestamps(context.Background())
+	tx, err := database.DB.SQL.BeginTx(ctx, nil)
+	if err != nil {
+		log.Errorf(log.DatabaseMgr, "Insert transaction being failed: %v", err)
+		return
+	}
+
+	if repository.GetSQLDialect() == database.DBSQLite3 {
+		err = insertSQLite(ctx, tx, []Details{in})
+	} else {
+		err = insertPostgresql(ctx, tx, []Details{in})
+	}
+
+	if err != nil {
+		log.Errorf(log.DatabaseMgr, "Insert failed: %v", err)
+		err = tx.Rollback()
+		if err != nil {
+			log.Errorf(log.DatabaseMgr, "Insert Transaction rollback failed: %v", err)
+		}
+		return
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		log.Errorf(log.DatabaseMgr, "Insert Transaction commit failed: %v", err)
+		err = tx.Rollback()
+		if err != nil {
+			log.Errorf(log.DatabaseMgr, "Insert Transaction rollback failed: %v", err)
+		}
+		return
+	}
+}
+
+// InsertMany writes multiple entries into databass
+func InsertMany(in []Details) error {
+	if database.DB.SQL == nil {
+		return database.ErrDatabaseSupportDisabled
+	}
+
+	ctx := boil.SkipTimestamps(context.Background())
+	tx, err := database.DB.SQL.BeginTx(ctx, nil)
+	if err != nil {
+		log.Errorf(log.DatabaseMgr, "Insert transaction being failed: %v", err)
+		return err
+	}
+
+	if repository.GetSQLDialect() == database.DBSQLite3 {
+		err = insertSQLite(ctx, tx, in)
+	} else {
+		err = insertPostgresql(ctx, tx, in)
+	}
+
+	if err != nil {
+		log.Errorf(log.DatabaseMgr, "Insert failed: %v", err)
+		err = tx.Rollback()
+		if err != nil {
+			log.Errorf(log.DatabaseMgr, "Insert Transaction rollback failed: %v", err)
+		}
+		return err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		log.Errorf(log.DatabaseMgr, "Insert Transaction commit failed: %v", err)
+		err = tx.Rollback()
```

Comment:
err -> errRB

Reply:
Have reworked these to pass back error as exchange/candle insert's return an error